src/class.h

```diff
@@ -12,14 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <vector>
+
+#include "../imageio/image_dec.h"
+#include "../imageio/imageio_util.h"
 #include "webp/encode.h"
+#include "webp/mux.h"
 
 namespace libwebp {
 
-// Class that does something.
+// Takes time stamped images as an input and produces an animation.
 class Thumbnailer {
-  // TODO: add an API to add frames with a timestamp.
-  // TODO: add an API to generate the animation.
+ private:
+  WebPAnimEncoder* enc = NULL;
+  std::vector<WebPPicture> pics;
+  WebPData webp_data;
+  WebPAnimEncoderOptions anim_config;
+  WebPConfig config;
+  std::vector<int> timestamps;
+  int loop_count;
+  int num_pic = 0;
+
+ public:
+  Thumbnailer();
+  ~Thumbnailer();
+
+  // Add frame with a timestamp.
+  bool AddFrame(const WebPPicture& pic, int timestamp);
+
+  // Generate the animation.
```

Comment:
Please conjugate action in description (it is a Google style): Generates the animation.
(same above with "Adds frame").